File: Data Structures & Algorithms/is-anagram/submission-0.cpp
#include <vector>
#include <unordered_set>
#include <string> 

using namespace std;

class Solution {
public:
    bool isAnagram(string s, string t) {

        if (s.length() != t.length()) {
            return false;
        }

        std::vector<int> count(26, 0);


        for (int i = 0; i < s.length(); i++) {
            count[s[i] - 'a']++;
            count[t[i] - 'a']--;
        }
        

        for (int k = 0; k < count.size(); k++) {
            if (count[k] != 0) {
                return false;
            }
        }
        return true;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
import heapq
from collections import Counter
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        result = []

        count = Counter(nums)

        for num, freq in count.items():
            result.append((-freq,num))


        heapq.heapify(result)
        ans = []

        for i in range(k):
            freq, num = heapq.heappop(result)
            ans.append(num)

        return ans
       



        